src/StudyUtils.in.jsm

```diff
@@ -76,52 +143,107 @@ function merge(source) {
+* @param {Object} args - query arguments, one or more object literal used to
+* build a query string
+* @returns {string} - an absolute url appended with a query string
+*/
 function mergeQueryArgs(url, ...args) {
-  /* currently left to right*/
+  // currently left to right
   // TODO, glind, decide order of merge here
   // TODO, use Object.assign, or ES7 spread
   const U = new URL(url);
+  // get the query string already attached to url, if it exists
   let q = U.search || "?";
+  // create an interface to interact with the query string
   q = new URLSearchParams(q);
-
   const merged = merge({}, ...args);
-
-  // get user info.
+  // Set each search parameter in "merged" to its value in the query string,
+  // building up the query string one search parameter at a time.
   Object.keys(merged).forEach((k) => {
+    // k, the search parameter (ex: fxVersion)
+    // q.get(k), returns the value of k, in query string, q (ex: 57.0.1a)
     log.debug(q.get(k), k, merged[k]);
     q.set(k, merged[k]);
   });
-
+  // append our new query string to the URL object made with "url"
   U.search = q.toString();
+  // return the full url, with the appended query string
   return U.toString();
 }
 
 // sampling utils
+/**
+* @async
+* Converts a string into its sha256 hexadecimal representation.
+* Note: This is ultimately used to make a hash of the user's telemetry clientID
+* and the study name.
+* @param {string} message - The message to convert.
+* @returns {string} hashHex - a hexadecimal, 256-bit hash
+*/
 async function sha256(message) {
-  const msgBuffer = new TextEncoder("utf-8").encode(message); // encode as UTF-8
-  const hashBuffer = await crypto.subtle.digest("SHA-256", msgBuffer); // hash the message
-  const hashArray = Array.from(new Uint8Array(hashBuffer)); // convert ArrayBuffer to Array
-  const hashHex = hashArray.map(b => ("00" + b.toString(16)).slice(-2)).join(""); // convert bytes to hex string
+  // encode as UTF-8
+  const msgBuffer = new TextEncoder("utf-8").encode(message);
+  // hash the message
+  const hashBuffer = await crypto.subtle.digest("SHA-256", msgBuffer);
+  // convert ArrayBuffer to Array
+  const hashArray = Array.from(new Uint8Array(hashBuffer));
+  // convert bytes to hex string
+  const hashHex
+    = hashArray.map(b => ("00" + b.toString(16)).slice(-2)).join("");
   return hashHex;
 }
 
+/**
+* Converts an array of length N into a cumulative sum array of length N,
+* where n_i = sum(array.slice(0,i)) i.e. each element is the sum of all
+* elements up to and including that element
+* This is ultimately used for turning sample weights (AKA weightedVariations)
+* into right hand limits (>= X) to  deterministically select which variation
+* a user receives.
+* @example [.25,.3,.45] => [.25,.55,1.0]; if a user's sample weight were .25,
+* they would fall into the left-most bucket
+* @param {Number[]} arr - An array of sample weights [0, 1)
+* @returns {Number[]} - A cumulative sum array of sample weights [0, 1)
```

Comment:
Should these be `[0, 1)`, or `[0, 1]` (or `(0, 1)`)?
Seems like an inconsistent bracket pairing.

(I think there were 5 other instances of `[0, 1)` in this PR, so it may be a search-replace thing.)

Reply:
This is a shorthand way of saying `0 <= value < 1`; these values primarily come from [Math.random()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random).

I've replaced `[0,1)` with the longer form `0 <= value < 1` to avoid confusion. In one of the cases (the output of `cumsum`), the value can actually equal 1, so I've updated that too.